flink-formats/flink-avro-glue-schema-registry/src/main/java/org/apache/flink/formats/avro/glue/schema/registry/GlueSchemaRegistryAvroFormatFactory.java

```diff
@@ -0,0 +1,162 @@
+import org.apache.flink.api.common.serialization.DeserializationSchema;
+import org.apache.flink.api.common.serialization.SerializationSchema;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.configuration.ReadableConfig;
+import org.apache.flink.formats.avro.AvroRowDataDeserializationSchema;
+import org.apache.flink.formats.avro.AvroRowDataSerializationSchema;
+import org.apache.flink.formats.avro.AvroToRowDataConverters;
+import org.apache.flink.formats.avro.RowDataToAvroConverters;
+import org.apache.flink.formats.avro.typeutils.AvroSchemaConverter;
+import org.apache.flink.table.connector.ChangelogMode;
+import org.apache.flink.table.connector.format.DecodingFormat;
+import org.apache.flink.table.connector.format.EncodingFormat;
+import org.apache.flink.table.connector.sink.DynamicTableSink;
+import org.apache.flink.table.connector.source.DynamicTableSource;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.factories.DeserializationFormatFactory;
+import org.apache.flink.table.factories.DynamicTableFactory;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.flink.table.factories.SerializationFormatFactory;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.table.types.logical.RowType;
+
+/**
+ * Table format factory for providing configured instances of AWS Glue Schema
+ * Registry Avro to RowData {@link SerializationSchema} and
+ * {@link DeserializationSchema}.
+ */
+@Internal
+public class GlueSchemaRegistryAvroFormatFactory implements DeserializationFormatFactory, SerializationFormatFactory {
+    public static final String IDENTIFIER = "avro-glue";
+
+    @Override
+    public DecodingFormat<DeserializationSchema<RowData>> createDecodingFormat(DynamicTableFactory.Context context,
+            ReadableConfig formatOptions) {
+        FactoryUtil.validateFactoryOptions(this, formatOptions);
+        final Map<String, Object> configMap = buildConfigMap(formatOptions);
+
+        return new DecodingFormat<DeserializationSchema<RowData>>() {
+            @Override
+            public DeserializationSchema<RowData> createRuntimeDecoder(DynamicTableSource.Context context,
+                    DataType producedDataType) {
+                final RowType rowType = (RowType) producedDataType.getLogicalType();
+                final TypeInformation<RowData> rowDataTypeInfo = context.createTypeInformation(producedDataType);
+                return new AvroRowDataDeserializationSchema(
+                        GlueSchemaRegistryAvroDeserializationSchema
+                                .forGeneric(AvroSchemaConverter.convertToSchema(rowType), configMap),
+                        AvroToRowDataConverters.createRowConverter(rowType), rowDataTypeInfo);
+            }
+
+            @Override
+            public ChangelogMode getChangelogMode() {
+                return ChangelogMode.upsert();
```

Comment:
What does this mode refer to? Does this impact the type of connector? For instance, Kafka which supports upsert, and kinesis which does not (insert only)?

Reply:
I'm actually a bit confused on this point.  The `Format` interface requires implementation of the `getChangelogMode` method, but I don't understand why this is tied to the format at all.  If I'm using a Kafka-Upsert connector then I should get upsert records... if I'm using a generic Kafka stream I should expect all kinds of records, both of which are totally orthogonal to the format used. 

When I look at the other Format implementations they all seem to either be using the explicit form of `ChangelogMode.all()` or `ChangelogMode.insetOnly()` with no obvious (to me) pattern for why a given Format implementation gets one or the other.

I've changed the code in these two instances to `ChangelogMode.all()` to match the subset of other formats that also does this, including the `DebeziumAvroFormatFactory`, but without a deeper understanding of how the `Format.getChangelogMode` is used, I can't reason more precisely about what this _should_ be.

Reply:
The format can define a changelog, but it'll be up to the connector to defer to the format for providing the changelog mode. `SocketDynamicTableSource` in `flink-examples` would be an example where the connector just defers this to the format.

Reply:
So, I did a test of an actual flink job last night and discovered that unless this is set to `insertOnly` you can't create an upsert-kafka connector using it.  So I clearly don't understand what this changelog mode refers to, but I'm just going to cope.